Buildings/Experimental/DHC/Plants/Combined/Controls/BaseClasses/CoolingTowerLoop.mo

```diff
@@ -0,0 +1,529 @@
+  Buildings.Controls.OBC.CDL.Continuous.MovingAverage mea(delta=5*60)
+    "Moving mean"
+    annotation (Placement(transformation(extent={{-130,30},{-110,50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract delTem1 "Compute deltaT"
+    annotation (Placement(transformation(extent={{-100,60},{-80,80}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add lifPlu "Add target lift"
+    annotation (Placement(transformation(extent={{-130,90},{-110,110}})));
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter subApp(final p=-
+        dTHexCoo_nominal) "Substract HX approach"
+    annotation (Placement(transformation(extent={{-70,60},{-50,80}})));
+  Buildings.Controls.OBC.CDL.Continuous.Switch TSupSetUnb(y(unit="K",
+        displayUnit="degC"))
+    "Compute tower supply temperature setpoint, unbounded" annotation (
+      Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-20,260})));
+  Buildings.Controls.OBC.CDL.Integers.Sources.Constant modRej(
+    final k=Buildings.Experimental.DHC.Plants.Combined.Controls.ModeCondenserLoop.heatRejection)
+    "Heat rejection mode index"
+    annotation (Placement(transformation(extent={{-150,230},{-130,250}})));
+  Buildings.Controls.OBC.CDL.Integers.Equal isModRej "Heat rejection mode"
+    annotation (Placement(transformation(extent={{-110,250},{-90,270}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant setOth[2](final k=
+        TTanSet[:, 1] .- dTHexCoo_nominal)
+    "Target setpoint in any mode other than heat rejection"
+    annotation (Placement(transformation(extent={{-110,190},{-90,210}})));
+  Buildings.Controls.OBC.CDL.Routing.RealExtractor extSet(final nin=2)
+    "Extract setpoint value based on current mode"
+    annotation (Placement(transformation(extent={{-70,230},{-50,250}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiplyByParameter ratDes(final k=abs(1
+        /QChiWat_flow_nominal)) "Ratio to design capacity"
+    annotation (Placement(transformation(extent={{-160,150},{-140,170}})));
+  Buildings.Controls.OBC.CDL.Continuous.Line lif "Compute target chiller lift"
+    annotation (Placement(transformation(extent={{-110,130},{-90,150}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant xLif[2](k={0.1,1})
+    "x-value for lift reset"
+    annotation (Placement(transformation(extent={{-160,180},{-140,200}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant yLif[2](final k={
+        dTLifChi_min,dTLifChi_nominal}) "y-value for lift reset"
+    annotation (Placement(transformation(extent={{-160,120},{-140,140}})));
+  StagingPump staPum(
+    nPum=nPumConWatCoo,
+    have_flowCriterion=false,
+    yDow=0.4,
+    yUp=0.8)
+    "Stage pumps"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold cmpFlo(t=3, h=3e-3)
+    "Flow criterion to enable lead pump"
```

Comment:
@mwetter I addressed this and other instances as we discussed in commits https://github.com/lbl-srg/modelica-buildings/pull/3263/commits/1fc7b11c1974f398a562956c72cd201faf6abfe1 through https://github.com/lbl-srg/modelica-buildings/pull/3263/commits/41a7c3177183734a75bb3bb16d853fe8e50170f7.  Note that the threshold and hysteresis parameters of the greater/less than CDL block can't be dependent on iteration variables of the initialization problem, so [here](https://github.com/lbl-srg/modelica-buildings/pull/3263/commits/b6ca2abfe5a55d72691ec556bfdf626edd1fbd66#diff-6b12c02182a8e2551902e7fbf229c7c7b0bb5a9d3ad583a5bdeb16437f6e8758R216) I needed to use a hard-coded cp and [here](https://github.com/lbl-srg/modelica-buildings/pull/3263/commits/b6ca2abfe5a55d72691ec556bfdf626edd1fbd66#diff-bfb29305f9a33a5b446e9b82a281a718287d8d10715bcb5ddc6c071fb0030221R235) I needed to introduce an approximate value of the nominal cooling capacity of the heat recovery chillers in cascade cooling mode which is directly specified (basically as datChiHea.QEva_flow_nominal), instead of the exact value that is found through iteration at the initialization.